lib/puppet/provider/reboot/windows.rb

```diff
@@ -2,8 +2,8 @@
 require 'open3'
 
 Puppet::Type.type(:reboot).provide :windows do
-  confine :operatingsystem => :windows
-  defaultfor :operatingsystem => :windows
+  confine :kernel => :windows
+  defaultfor :kernel => :windows
```

Comment:
Sorry for the late response on this, but I was out of the office for awhile. It looks like this change made it into the master branch even though this PR was only adding Linux support. Can we make sure to revert this part of the PR to the windows provider?